---
title: "Data analysis"
author: "Aoyue Bi, Daxing Xu"
date: "`r Sys.Date()`"
output: 
  html_document: # html_notebook, html_document
    toc: true
    toc_float:
      toc_collapsed: true
    toc_depth: 3
    number_sections: true
    theme: simplex # default, united, cerulean, journal, flatly, darkly, readable, spacelab, united, cosmo, lumen, paper, sandstone, simplex, and yeti
    highlight: haddock # default, tango, pygments, kate, monochrome, espresso, zenburn, haddock, and breezedark
    # bibliography: bibliography.json
---

### ********************
## **项目分析相关
### ********************
### taxaInfoDB 根据VcfID添加倍性及亚群分组信息
```{r}
### 导入种质数据库,并选取特定的列
dftaxaDB <- read.delim("/Users/Aoyue/project/wheatVMap2_1000/001_germplasm/019_WheatVMap2_GermplasmInfo.txt") %>% select(Taxa=VcfID,GenomeType)

### 添加列信息
dfdel_addTaxainfo <- dfdel %>% left_join(.,dftaxaDB,by="Taxa")
```

### chrDB 根据染色体号添加亚基因组信息，并过滤四倍体中的D亚基因组数据
```{r}
library(tidyverse)
# ### 导入染色体-亚基因组数据库，并选取特定的列
dfchrDB <- read.delim("/Users/Aoyue/project/wheatVMap2_1000/002_dataAnalysis/000_chrDB/chrConvertionRule_byAoyue.txt") %>% select(Chr = ChrID,Subgenome)

### 添加列信息
dfdel_addTaxainfo_addChrDB <- dfdel_addTaxainfo %>% left_join(.,dfchrDB,by="Chr")
### 过滤某些多余的行内容
dfdel_addTaxainfo_addChrDB2 <- dfdel_addTaxainfo_addChrDB %>% 
  mutate(MergedGenomeTypeSub = paste(GenomeType,"_",Subgenome,sep = "")) %>% 
  filter(MergedGenomeTypeSub != "AABB_D", MergedGenomeTypeSub != "DD_A",MergedGenomeTypeSub != "DD_B") %>% 
  select(-MergedGenomeTypeSub)

```

### taxaInfoDB 添加 Continent_forTree 分类
```{r}
dataFile <- c("/Users/Aoyue/project/wheatVMap2_1000/001_germplasm/011_WheatVMap2_GermplasmInfo_20210725.txt")

### 添加新的建树分组
# Part_Continent <- levels(as.factor(df1062$Part_Continent))
Part_Continent <- c("Africa","Central Asia","Central Europe","East Asia","Eastern Europe","North America","Northern Europe","Oceania","South America","South Asia","Southeast Europe","Southern Europe","Western Asia","Western Europe")
Continent_forTree <- c("Africa","Central and South Asia","Europe","East Asia","Europe","America","Europe","Oceania","America","Central and South Asia","Europe","Europe","Western Asia","Europe")
dfhash <- data.frame(Part_Continent,Continent_forTree)

df <- read.delim(dataFile) %>% 
  left_join(.,dfhash,by="Part_Continent") 

write.table(df,file="~/Documents/hash.txt",quote=F,col.name=T,row.names=F,sep = "\t") ### example

```
### Load 分类
```{r}
factorA <- c("Ratio_002_nonsynonymous","Ratio_003_nonsynGERPandDerivedSIFT","Ratio_004_nonsynDerivedSIFT","Ratio_005_GERP","Ratio_006_StopGain","Ratio_007_VEP","Ratio_008_snpEff","Ratio_009_VEP_stopGained","010_GERP16way","011_GERP16way_1.2_max","012_GERP16way_1.2_2.5","013_GERP16way_2.5_max")

labelA <- c("Nonsynonymous","GERP+SIFT","SIFT","GERP","StopGain_SIFT","VEP","snpEff","StopGain_VEP","GERP16way","GERP16way1.2","GERP16way1.2_2.5","GERP16way2.5")

```


### ********************
## **颜色形状哈希相关
### ********************

### 显示自己挑选的几种颜色
```{r}
colB <- brewer.pal(12,"Set1")[c(1:12)]
colB <- brewer.pal(8,"Dark2")[c(1:8)]

df <- data_frame(x=colB)
df$x <- factor(df$x,levels = df$x)
ggplot(df, aes(x=x, y=1, fill=x)) +
  geom_bar(stat="identity")+theme_minimal()+
  scale_fill_manual(values = colB)

```
### A B D 亚基因组
```{r}
subA <- c("A","B","D")
colB <- c("#fd8582","#967bce","#4bcdc6")

df <- data_frame(x=colB)
df$x <- factor(df$x,levels = df$x)
ggplot(df, aes(x=x, y=1, fill=x)) +
  geom_bar(stat="identity")+theme_minimal()+
  scale_fill_manual(values = colB)

```
### GenomeType 六四二倍体
```{r}
subA <- c("AABB","AABBDD","DD")
colB <- c('#ffd702','#fc6e6e',"#87cef9")

df <- data_frame(x=colB,y=subA)
df$x <- factor(df$x,levels = df$x)
ggplot(df, aes(x=x, y=1, fill=x)) +
  geom_bar(stat="identity")+theme_minimal()+
  scale_fill_manual(values = colB)

```


### 亚群- 六四二642倍体在一起
```{r}
### 不包括OtherTetraploids
factorA <- c("Strangulata","Wild_emmer","Domesticated_emmer","Free_threshing_tetraploids", "Landrace","Cultivar")

labelsA <- c("Strangulata","Wild emmer","Domesticated emmer","Free-threshing tetraploids", "Landrace","Cultivar")

colB <- c("#87cef9","#ffd702","#7f5701","#016699", "#fc6e6e","#9900ff")

dfcol <- data_frame(x=factorA,y=labelsA,z=colB) 
### 关键：设置好因子的顺序
dfcol$x <- factor(dfcol$x,levels = dfcol$x,labels = dfcol$y)
ggplot(dfcol, aes(x=x, y=1, fill=x)) +
  # geom_bar(stat="identity",alpha = 0.9)+theme_minimal()+
  geom_point(shape=21,color="gray",size=2)+
  scale_fill_manual(values = colB)+
  theme_void()

### 包括Other类型
factorA <- c("Wild_emmer","Domesticated_emmer","Free_threshing_tetraploids","OtherTetraploids", "Landrace","Cultivar","OtherHexaploids","Strangulata")

labelsA <- c("Wild emmer","Domesticated emmer","Free-threshing tetraploids","OtherTetraploids","Landrace","Cultivar","OtherHexaploids","Strangulata")

colB <- c("#ffd702","#7f5701","#016699","#00f3ff", "#fc6e6e","#9900ff","#fe63c2","#87cef9")

dfcol <- data_frame(x=factorA,y=labelsA,z=colB) 
### 关键：设置好因子的顺序
dfcol$x <- factor(dfcol$x,levels = dfcol$x,labels = dfcol$y)

p <- ggplot(dfcol, aes(x=x, y=1, fill=x)) +
  # geom_bar(stat="identity",alpha = 0.9)+theme_minimal()+
  geom_point(shape=21,color="gray",size=2)+
  scale_fill_manual(values = colB)+
  theme_void()

p

ggsave(p,filename = "~/Documents/legend.pdf", height = 3, width = 3)
```
### 亚群- 六四倍体64在一起
```{r}
factorA <- c("Wild_emmer","Domesticated_emmer","Free_threshing_tetraploids","OtherTetraploids", "Landrace","Cultivar","OtherHexaploids")

labelsA <- c("Wild emmer","Domesticated emmer","Free-threshing tetraploids","OtherTetraploids", "Landrace","Cultivar","OtherHexaploids")

colB <- c("#ffd702","#7f5701","#016699","#00f3ff", "#fc6e6e","#9900ff","#fe63c2")

dfcol <- data_frame(x=factorA,y=labelsA,z=colB) 
### 关键：设置好因子的顺序
dfcol$x <- factor(dfcol$x,levels = dfcol$x,labels = dfcol$y)
p1 <- ggplot(dfcol, aes(x=x, y=1, fill=x)) +
  geom_bar(stat="identity",alpha = 0.9)+theme_minimal()+
  scale_fill_manual(values = colB)
p1
```
### 亚群- 六二倍体62在一起
```{r}
factorA <- c("Strangulata","Landrace","Cultivar","OtherHexaploids")

labelsA <- c("Strangulata","Landrace","Cultivar","OtherHexaploids")

colB <- c("#87cef9","#fc6e6e","#9900ff","#fe63c2")

dfcol <- data_frame(x=factorA,y=labelsA,z=colB) 
### 关键：设置好因子的顺序
dfcol$x <- factor(dfcol$x,levels = dfcol$x,labels = dfcol$y)
ggplot(dfcol, aes(x=x, y=1, fill=x)) +
  geom_bar(stat="identity",alpha = 0.9)+theme_minimal()+
  scale_fill_manual(values = colB)
```

### 大洲(树)
```{r}
continent <- c("Africa", "America","Central and South Asia","East Asia","Europe","Oceania","Western Asia")  
colB <- c("#7B241C",'#dbb3ff','#A9A9A9','#fc6e6e','#FF9900','#7dbde8','#82C782')
shapeB <- c(16,8,18,6,15,7,17)

df <- data_frame(x=continent,y=1,colB,shapeB)

ggplot(df,aes(x=x,y=y))+
  geom_point(aes(shape=x,color=x),size=4)+
  scale_shape_manual(values = shapeB)+
  scale_color_manual(values = colB)+
  theme_minimal()
```

### 亚群六倍体_11
```{r}
### 六倍体的亚群
factorA <- c("Landrace","Cultivar","OtherHexaploids","compactum","sphaerococcum","tibeticum","vavilovii","petropavlovskyi","yunna-nense","macha","spelta")

labelsA <- c("Landrace","Cultivar","OtherHexaploids","Club wheat","Indian dwarf wheat","Tibetan semi-wild","Vavilovii","Xinjiang wheat","Yunan wheat","Macha","Spelt")

# colotherHexaploid <- brewer.pal(12,"Set1")[1:7]
# colB <- c("#fc6e6e","#9900ff","#fe63c2","gray",colotherHexaploid)
colB <- c("#fc6e6e","#9900ff","#fe63c2","gray","#E41A1C","#984EA3","#FF7F00","#FFFF33","#A65628","#377EB8","#4DAF4A")

dfcol <- data_frame(x=factorA,y=labelsA,z=colB) 
### 关键：设置好因子的顺序
dfcol$x <- factor(dfcol$x,levels = dfcol$x,labels = dfcol$y)
ggplot(dfcol, aes(x=x, y=1, fill=x)) +
  geom_bar(stat="identity",alpha = 0.9)+theme_minimal()+
  scale_fill_manual(values = colB)

```
### 亚群四倍体_9
```{r}
### 四倍体的亚群
factorA <- c("dicoccoides","dicoccum","durum","turanicum","polonicum","turgidum","karamyschevii","ispahanicum","carthlicum")

labelsA <- c("Wild emmer","Domesticated emmer","Durum","Khorasan wheat ","Polish wheat","Rivet wheat","Georgian wheat","Ispahanicum","Persian wheat")

colB <- c("#ffd702","#7f5701","#016699","#66A61E","#E7298A","#666666","#7570B3","#D95F02","cyan") ### 色系来自 Dark2

dfcol <- data_frame(x=factorA,y=labelsA,z=colB) 
### 关键：设置好因子的顺序
dfcol$x <- factor(dfcol$x,levels = dfcol$x,labels = dfcol$y)
ggplot(dfcol, aes(x=x, y=1, fill=x)) +
  geom_bar(stat="identity",alpha = 0.9)+theme_minimal()+
  scale_fill_manual(values = colB)
```

### 亚群21
```{r}
factorA <- c("Landrace","Cultivar","OtherHexaploids","compactum","sphaerococcum","tibeticum","vavilovii","petropavlovskyi","yunna-nense","macha","spelta","dicoccoides","dicoccum","durum","turanicum","polonicum","turgidum","karamyschevii","ispahanicum","carthlicum","strangulata")

labelsA <- c("Landrace","Cultivar","OtherHexaploids","Club wheat","Indian dwarf wheat","Tibetan semi-wild","Vavilovii","Xinjiang wheat","Yunan wheat","Macha","Spelt","Wild emmer","Domesticated emmer","Durum","Khorasan wheat ","Polish wheat","Rivet wheat","Georgian wheat","Ispahanicum","Persian wheat","Strangulata")

dfhash <- data_frame(Subspecies_by22_TreeValidated=factorA,Subspecies22=labelsA) 

```

### Landrace 按照大洲区分
```{r}
### 六倍体的Landrace按照大洲
factorA <- c()

labelsA <- c()

colB <- c()

dfcol <- data_frame(x=factorA,y=labelsA,z=colB) 
### 关键：设置好因子的顺序
dfcol$x <- factor(dfcol$x,levels = dfcol$x,labels = dfcol$y)
ggplot(dfcol, aes(x=x, y=1, fill=x)) +
  geom_bar(stat="identity",alpha = 0.9)+theme_minimal()+
  scale_fill_manual(values = colB)
```

### 基因型热图
```{r}
factorA <- c("0","1","2","9")

labelsA <- c("0/0","0/1","1/1","./.")

# colB <- c("#FEF9E7","#E8DAEF","#CD6155","#F2F3F4")
colB <- c("#FEF9E7","#85C1E9","#CD6155","#F2F3F4")

dfcol <- data_frame(x=factorA,y=labelsA,z=colB) 
### 关键：设置好因子的顺序
dfcol$x <- factor(dfcol$x,levels = dfcol$x,labels = dfcol$y)
ggplot(dfcol, aes(x=x, y=1, fill=x)) +
  geom_bar(stat="identity",alpha = 0.9)+theme_minimal()+
  scale_fill_manual(values = colB)
```
## Synonymous Nonsynonymous Delterious
```{r}
factorA <- c("Deleterious","Nonsynonymous","Synonymous")

labelsA <- c("Deleterious","Nonsynonymous","Synonymous")

colB <- c("#d5311c","#e69f00","#004680")

dfcol <- data_frame(x=factorA,y=labelsA,z=colB) 
### 关键：设置好因子的顺序
dfcol$x <- factor(dfcol$x,levels = dfcol$x,labels = dfcol$y)
ggplot(dfcol, aes(x=x, y=1, fill=x)) +
  geom_bar(stat="identity",alpha = 0.9)+theme_minimal()+
  scale_fill_manual(values = colB)

```







### ********************
## **代码技巧相关
### ********************

### 检查每列数据异同
```{r}
### 和达兴的数据框一列一列的核对
### 判断两列是否一致
dataFile <- c("/Users/Aoyue/project/wheatVMap2_1000/001_germplasm/010_WheatVMap2_GermplasmInfo_20210725.txt")
dfV1 <- read.delim(dataFile) %>% 
  select(VcfID,Group1=TreeValidatedGroupbySubspecies) 

dataFile <- c("/Users/Aoyue/project/wheatVMap2_1000/001_germplasm/GermplasmInfo_addChineseName_byDaxing.txt")
dfV2 <- read.delim(dataFile) %>% 
  select(VcfID,Group2=TreeValidatedGroupbySubspecies)

dfdiff <- dfV1 %>% left_join(.,dfV2,by="VcfID") %>% 
  mutate(IfEqual = if_else(as.character(Group1)  == as.character(Group2),1,0)) %>% 
  filter(IfEqual==0)
```

```{r}
### 导入种质数据库,并选取特定的列
dataFile <- c("/Users/Aoyue/project/wheatVMap2_1000/001_germplasm/011_WheatVMap2_GermplasmInfo_20210725.txt")
dftaxaDB <- read.delim(dataFile) %>% select(Taxa=VcfID,GenomeType, Subspecies) %>% 
  filter(GenomeType == "AABBDD") %>% group_by(Subspecies) %>% 
  count()
```

### 检查每个文件的行数，并打出log，这里只适用txt文件，不适用vcf文件
```{r}
inputDir <- c("/Users/Aoyue/project/wheatVMap2_1000/002_dataAnalysis/004_annoDB/007_geneVCF")
fsList <- list.files(inputDir)
length(fsList)
cnt <- 0
dflog <- NULL
for (i in fsList) {
  df <- read.delim(file.path(inputDir,i)) ### 读入数据框
  cnt <- cnt+nrow(df) ###统计总行数
  log <- paste(i, nrow(df),sep = "\t")
  dflog2 <- data.frame(filename = i,SNPNum = nrow(df))
  dflog <- rbind(dflog,dflog2)
  print(log)
}

print(cnt)
write.table(dflog,file="~/Documents/lineNumber.txt",quote=F,col.name=T,row.names=F,sep = "\t")

```
### 检查每列变量的因子类型，并打印
```{r}
df <- dfdel_long

### 一定一定记得加引号
aofactor <- c("VariantsGroup","GenomeType","Continent","Part_Continent","Subspecies","TreeValidatedGroupbySubspecies","Subgenome","DelCountGroup")

# dfcount <- df %>% summarise(across(.cols = aofactor,.fns = n_distinct))
### 统计每种分类下的因子分别是什么
l <- list() ### 先建立list
for(i in aofactor){
  d <- df[,i]
  d <- d[!is.na(d)]
  l[[i]] <- levels(as.factor(d))
}

fp <- paste("~/Documents/TaxaCategory",".txt",sep="") ### txt format
# fp <- paste("~/Documents/TaxaCategory",".csv",sep="") ### csv format
capture.output(l,file = fp)
```

### 将因子作为x轴并按顺序排序，并更改标签
- 操作步骤：①读取数据框，②一定记得先将因子排序；③在再数据库进行操作，统计因子的相关信息；④在使用scale_x_discrete 排序功能时，可以用labels将因子进行替代，更换成内容更多的信息
```{r}
### 导入种质数据库,并选取特定的列
dataFile <- c("data/001_TaxaDB/018_WheatVMap2_GermplasmInfo.txt")
dftaxaDB <- read.delim(dataFile) %>% 
  select(Taxa=VcfID,GenomeType, GroupbyContinent) %>% 
  filter(str_detect(GroupbyContinent,"LR") | str_detect(GroupbyContinent,"CL"))

dataFile <- c("data/002_pca/004_matrix_Asub.txt.gz")
dfmatrixA <- read.delim(dataFile) %>% select(Taxa,ABD_1142) %>% mutate(Sub = "A")
dataFile <- c("data/002_pca/005_matrix_Bsub.txt.gz") 
dfmatrixB <- read.delim(dataFile) %>% select(Taxa,ABD_1142) %>% mutate(Sub = "B")
dataFile <- c("data/002_pca/006_matrix_Dsub.txt.gz")
dfmatrixD <- read.delim(dataFile) %>% select(Taxa,ABD_1142) %>% mutate(Sub = "D")

df <- rbind(dfmatrixA,dfmatrixB,dfmatrixD) %>% 
  rename(IBSdistance2CS=ABD_1142) %>% 
  right_join(.,dftaxaDB,by="Taxa")

### 统计每个亚基因组内，以亚洲为单位的均值，按照B亚基因组排序
### dfV2 为排序后的亚洲列表
dfV1 <- df %>% group_by(Sub,GroupbyContinent) %>% summarise(mean = mean(IBSdistance2CS)) %>% arrange(.,Sub,-mean) %>% ungroup() %>%  filter(Sub=="B") %>% select(GroupbyContinent)

### 因子排序
df$GroupbyContinent <- factor(df$GroupbyContinent,levels = dfV1$GroupbyContinent)

### 计算每个亚群的个数
dftaxaCount <- df %>% group_by(Sub,GroupbyContinent) %>% 
  count() %>% ungroup() %>%  distinct(.,GroupbyContinent,.keep_all = T) %>% 
  select(-Sub) %>% mutate(GroupbyContinent_AddCount = str_c(GroupbyContinent," (", n, ")",sep = "")) %>% 
  select(-n)

### 阴影，discrete x 
p <- ggplot(df,aes(x=GroupbyContinent,y=IBSdistance2CS))+
  ylab("IBS distance to CS")+xlab("")+
  geom_rect(aes(xmin=1-0.5,xmax=1+0.5,ymin=-Inf,ymax=Inf),fill="#f2f2f2",alpha=0.5)+
  geom_rect(aes(xmin=3-0.5,xmax=3+0.5,ymin=-Inf,ymax=Inf),fill="#f2f2f2",alpha=0.5)+
  geom_rect(aes(xmin=5-0.5,xmax=5+0.5,ymin=-Inf,ymax=Inf),fill="#f2f2f2",alpha=0.5)+
  geom_rect(aes(xmin=7-0.5,xmax=7+0.5,ymin=-Inf,ymax=Inf),fill="#f2f2f2",alpha=0.5)+

  # stat_summary(fun=mean, geom="point", color="red",position = position_dodge(1),size=0.1)+
  stat_summary(aes(color=Sub),fun.data=mean_sdl,position = position_dodge(0.5),geom="pointrange",size=0.3)+
  scale_color_manual(values = c("#fd8582","#967bce","#4bcdc6"),name='')+
  scale_x_discrete(labels = dftaxaCount$GroupbyContinent_AddCount)+
  theme(axis.text.x= element_text(angle=45, vjust=1,hjust = 1))+
  # theme(plot.margin=unit(rep(0.3,4),'lines'))+
  theme(plot.margin=unit(rep(1.2,4),'lines'))+
  theme(
    panel.background = element_blank(),
    panel.border = element_rect(colour = "black", fill=NA, size=0.4),
    text = element_text(size = 12)
    # ,legend.position = 'none'
    ,legend.key = element_blank()
    )
p
ggsave(p,filename = "/Users/Aoyue/Documents/test.pdf",height = 3.43,width = 4)

```

### 添加新列，多文件处理。snpEff 和 VEP结果
#### 该程序功能：单独->每个文件单独输出处理后，单独输出
```{r}
inputDir <- c("")
outputDir <- c("")

fsList <- list.files(inputDir) ### 列出文件名字
length(fsList)
cnt <- 0
dflog <- NULL
for (i in fsList) {
  df <- read.delim(file.path(inputDir,i))
  chr <- str_sub(i,4,6)
  
  df <- df ### data operate

  
  cnt <- cnt+nrow(df)
  ### outtput file one by one 
  filenameout <- file.path(outputDir,str_replace(i,".txt.gz",".txt"))
  write.table(df,file=filenameout,quote=F,col.name=T,row.names=F,sep = "\t")
  ### log
  log <- paste(i, nrow(df),sep = "\t")
  dflog2 <- data.frame(filename = i, LineA = nrow(df))
  dflog <- rbind(dflog,dflog2)
  print(log) ### 打印当前文件的名字和行数
}

print(cnt) ### 打印总行数
write.table(dflog,file="~/Documents/SNPNum_vmap2.1_geneRegion.txt",quote=F,col.name=T,row.names=F,sep = "\t")
```

##### 经典例子
```{r}
inputDir <- c("/Users/Aoyue/project/wheatVMap2_1000/002_dataAnalysis/009_xpclr/003_summary_XPCLR/003_deleteriousXPCLR")
outputDir <- c("/Users/Aoyue/project/wheatVMap2_1000/002_dataAnalysis/009_xpclr/003_summary_XPCLR/004_from003_filterSub_sumSub")
fsList <- list.files(inputDir)
length(fsList)

cnt <- 0
dflog <- NULL

### 导入种质数据库,并选取特定的列
dataFile <- c("/Users/Aoyue/project/wheatVMap2_1000/001_germplasm/019_WheatVMap2_GermplasmInfo.txt")
dftaxaDB <- read.delim(dataFile) %>% select(Taxa=VcfID,GenomeType)

### 导入染色体-亚基因组数据库，并选取特定的列
dataFile <- c("/Users/Aoyue/project/wheatVMap2_1000/002_dataAnalysis/000_chrDB/chrConvertionRule_byAoyue.txt")
dfchrDB <- read.delim(dataFile) %>% select(Chr = ChrID,Subgenome)

for (i in fsList) {
  df1 <- read.delim(file.path(inputDir,i))
  df2 <- df1 %>% left_join(.,dftaxaDB,by="Taxa") %>% 
    left_join(.,dfchrDB,by="Chr") %>% 
    mutate(MergedGenomeTypeSub = paste(GenomeType,"_",Subgenome,sep = "")) %>% 
    filter(MergedGenomeTypeSub != "AABB_D", MergedGenomeTypeSub != "DD_A",MergedGenomeTypeSub != "DD_B") %>% 
    select(-MergedGenomeTypeSub) %>% 
    ###************ group_by ***********需要根据实际修改
    group_by(Taxa,Subgenome,VariantsGroup,IfSelected,Group_refobj) %>% ##求每个个体，每个亚基因组，每种类型的count之和
    summarise(TotalDerivedSNPCount = sum(TotalDerivedSNPCount),
              HomoDerivedSNPCount = sum(HomoDerivedSNPCount),
              HeterDerivedSNPCount = sum(HeterDerivedSNPCount),
              SiteCountWithMinDepth = sum(SiteCountWithMinDepth))
  
  outfileS <- file.path(outputDir,str_replace(i,".gz",""))
  write.table(df2,file=outfileS,quote=F,col.name=T,row.names=F,sep = "\t")
  
  cnt <- cnt+nrow(df2)
  ### log
  log <- paste(i, nrow(df2),sep = "\t")
  dflog2 <- data.frame(filename = i, LineA = nrow(df2))
  dflog <- rbind(dflog,dflog2)
  print(log) ### 打印当前文件的名字和行数
}
```

#### 该程序功能： 合并->每个文件单独输出处理后，合并起来输出
```{r}
inputDir <- c("")
outDir <- c("")
dfoutM <- NULL
fsList <- list.files(inputDir) ### 列出文件名字
length(fsList)
cnt <- 0
dflog <- NULL
for (i in fsList) {
  df <- read.delim(file.path(inputDir,i))
  chr <- str_sub(i,4,6)
  
  dfout <- df    ### data operate

  
  ### output file into one 
  dfoutM <- rbind(dfoutM,dfout)
  ### log
  cnt <- cnt+nrow(dfout)
  log <- paste(i, nrow(dfout),sep = "\t")
  dflog2 <- data.frame(filename = i, LineA = nrow(dfout))
  dflog <- rbind(dflog,dflog2)
  print(log) ### 打印当前文件的名字和行数
}

print(cnt) ### 打印总行数
write.table(dflog,file="~/Documents/log.txt",quote=F,col.name=T,row.names=F,sep = "\t")
# ### output result
dfoutM <- dfoutM %>% select()
write.table(dfoutM,file="~/Documents/test.txt",quote=F,col.name=T,row.names=F,sep = "\t")
```
#### 该程序功能： 多文件夹->每个文件单独输出处理后，合并起来输出
```{r}
inputDir <- c("/Users/Aoyue/project/wheatVMap2_1000/002_dataAnalysis/004_annoDB/004_SNP_Annotation")
outDir <- c("/Users/Aoyue/project/wheatVMap2_1000/002_dataAnalysis/004_annoDB/005_SNP_Annotation")
vepDir <- c("/Users/Aoyue/project/wheatVMap2_1000/002_dataAnalysis/004_annoDB/000_vep")
snpEffDir <- c("/Users/Aoyue/project/wheatVMap2_1000/002_dataAnalysis/004_annoDB/000_snpEff")

fsList <- list.files(inputDir) ### 列出文件名字
length(fsList)
cnt <- 0
dflog <- NULL
for (i in fsList) {
  df <- read.delim(file.path(inputDir,i))
  chr <- str_sub(i,4,6)
  
  vepFileS <- file.path(vepDir,paste("chr",chr,"_gene_vmap2.1.vepAnn.variant.effect.txt.gz",sep = ""))
  dfvep <- read.delim(vepFileS) %>% select(-Gene) %>%  rename(Effect_VEP=Consequence, Impact_VEP = Impact)
  
  snpEffFileS <- file.path(snpEffDir,paste("chr",chr,"_gene_vmap2.1.ann.effect.txt.gz",sep = ""))
  dfsnpEff <- read.delim(snpEffFileS) %>% 
    select(-Gene) %>% 
    rename(Effect_snpEff = Effect,Impact_snpEff = Impact) %>% 
    mutate(unionA=str_c(Chr,Pos,Transcript,sep = "\t")) %>% 
    distinct(.,unionA,.keep_all = T) %>% 
    select(-unionA)
  
  df <- df %>%
    left_join(.,dfvep,by=c("Chr","Pos","Transcript")) %>% 
    left_join(.,dfsnpEff,by=c("Chr","Pos","Transcript")) %>% 
    
  
  cnt <- cnt+nrow(df)
  ### outtput
  filenameout <- file.path(outDir,str_replace(i,".txt.gz",".txt"))
  write.table(df,file=filenameout,quote=F,col.name=T,row.names=F,sep = "\t")
  
  log <- paste(i, nrow(df),sep = "\t")
  dflog2 <- data.frame(filename = i,SNPNum = nrow(df))
  dflog <- rbind(dflog,dflog2)
  print(log)
}

print(cnt)
write.table(dflog,file="~/Documents/SNPNum_vmap2.1_geneRegion.txt",quote=F,col.name=T,row.names=F,sep = "\t")

```

### ********************
## **函数相关
### ********************
### Aoyue's function
```{r}
aolevels <- function(df,col){
  aovector <- df[,col]
  aovector <- aovector[!is.na(aovector)] 
  aoFactor <- levels(as.factor(aovector))
  # print(aoFactor)
  return(aoFactor)
}

aolevels(df,"GenomeType")
```


### Pad a string
https://stringr.tidyverse.org/reference/str_pad.html

### replace_na()
- 实现用同一个值替换一列中的所有 NA 值，该函数接受一个命名列表，其成分为“列名=替换值”
```{r}
starwars %>%
  replace_na(list(hair_color = "UNKNOWN",
                  birth_year = 99999))
```

### rowwise() 
- rowwise() is a function from dplyr that groups the data-frame row-wise, that is each row is a group

### across()
```{r}
dfdelCount <- read.delim("/Users/Aoyue/project/wheatVMap2_1000/002_dataAnalysis/009_xpclr/004_summary_XPCLR_top005/006_merge/delCount.txt")

# dfdelCount <- read.delim("")

### 导入种质数据库,并选取特定的列
dataFile <- c("/Users/Aoyue/project/wheatVMap2_1000/001_germplasm/019_WheatVMap2_GermplasmInfo.txt")
dftaxaDB <- read.delim(dataFile) %>% select(Taxa=VcfID,GroupbyContinent,IfOnVmap2_S643,ReleasedIntroducedYear) %>% 
  mutate(ReleasedIntroducedYear = str_replace(ReleasedIntroducedYear,"s",""))

df <- dfdelCount %>% 
  left_join(.,dftaxaDB,by="Taxa") %>% 
  # filter(IfSelected==1) %>% 
  filter(GroupbyContinent == "CL") %>% 
  filter(!is.na(ReleasedIntroducedYear)) %>% 
  #### 可以选择哪一列作为有害突变的计数 ###
  # filter(VariantsGroup %in% c("002_nonsynonymous")) %>%
  # filter(VariantsGroup %in% c("007_VEP")) %>%
  # filter(VariantsGroup %in% c("003_nonsynGERPandDerivedSIFT")) %>%
  # filter(VariantsGroup %in% c("005_GERP")) %>%
  filter(VariantsGroup %in% c("002_nonsynonymous", "007_VEP")) %>% 
  ### 不区分亚基因组
  group_by(Taxa,ReleasedIntroducedYear,VariantsGroup) %>% 
  summarise(across(.cols = c(TotalDerivedSNPCount:SiteCountWithMinDepth), .fns = sum)) %>% 
  ### 划分 Bin
  mutate(ReleasedIntroducedYear = as.numeric(ReleasedIntroducedYear)) %>%   
  # mutate(bin=cut_width(ReleasedIntroducedYear,width = 30,boundary = 0,closed = "left")) %>%
  # mutate(bin=cut(ReleasedIntroducedYear,breaks=c(1950,1980,2020))) 
  # mutate(bin=cut(ReleasedIntroducedYear,breaks=c(1950,1960,1970,1980,1990,2000,2020))) 
  mutate(bin=cut(ReleasedIntroducedYear,breaks=c(1950,1980,2000,2020))) 

  # group_by(bin) %>% count()
 
  # mutate(binString =  str_split_fixed(bin,",",n=2)[,1]) %>% 
  # mutate(binString =as.numeric(str_sub(binString,2)))
  # arrange(.,by=binString) 

### plot
library(RColorBrewer)
colB <- colorRampPalette(brewer.pal(9,"Reds")[c(3,5,7,9)])(5)

p <- df %>% 
  ggplot(aes(x=bin, y= TotalDerivedSNPCount)) +
  stat_boxplot(geom = "errorbar",width=0.05,position = position_dodge(1))+ # add error bar
  geom_boxplot(aes(fill=bin),alpha = 0.2,outlier.color = NA) +
  geom_point(position = position_jitterdodge(0.6),alpha = 0.8,aes(color=bin),size=2)+
  stat_summary(fun=mean, geom="point", color="yellow",position = position_dodge(0.8),size=0.5)+
  # stat_summary(fun.data=mean_sdl,position = position_dodge(0.5),geom="pointrange",size=0.3,color="orange")+
  # stat_summary(aes(group =  1), fun = "mean", geom = "line",color="gray") +
  scale_fill_manual(values = colB,name='')+  # scale_fill_manual(values = colB)+
  scale_color_manual(values = colB,name='')+  # scale_fill_manual(values = colB)+
  # scale_x_discrete(labels=c("1950s-1980s","1980s-1990s","1990s-2000s","2000s-2020s"))+
  scale_x_discrete(labels=c("1950s-1980s","1980s-2000s","2000s-2020s"))+
  # labs(x="",y="Nonsynonymous count")+
  # labs(x="",y="VEP count")+
  labs(x="",y="SNP count")+
  facet_wrap(~VariantsGroup,scales = "free")+
  theme(axis.text.x = element_text(vjust = 1, hjust = 1, angle = 45))+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        panel.background = element_blank(),
        axis.line = element_line(size=0.4, colour = "black"),
        text = element_text(size = 10),legend.position = 'none')
p
ggsave(p,filename = "~/Documents/test.pdf",height = 3,width = 4)
```


## !!!!??????? 答疑区
### 1. 色盘 distiller fermenter
The distiller scales extend brewer scales by smoothly interpolating 7 colours from any palette to a continuous scale. (连续的)
The fermenter scales provide binned versions of the brewer scales. （变成 Bin）

### ==================
```{r}
rm(list = setdiff(ls(), "df"))
```












 



