---
title: "Data analysis"
author: "Aoyue Bi, Daxing Xu"
date: "`r Sys.Date()`"
output: 
  html_document: # html_notebook, html_document
    toc: true
    toc_float:
      toc_collapsed: true
    toc_depth: 3
    number_sections: true
    theme: simplex # default, united, cerulean, journal, flatly, darkly, readable, spacelab, united, cosmo, lumen, paper, sandstone, simplex, and yeti
    highlight: haddock # default, tango, pygments, kate, monochrome, espresso, zenburn, haddock, and breezedark
    # bibliography: bibliography.json
---

# Fig3

```{r setup, echo =F, message = T, warning = F}
library(tidyverse)
# library(RColorBrewer)
# library(ggpubr)
# library(egg)
## eval 代码要不要运行
## echo 代码要不要输出
## include 图要不要输出
## warning 警告要不要输出
## message 默认Bin等信息要不要输出
```

#### =================
## Del load ratio
```{r}
choiceA <- "Ratio_031_GERP16way_1.5andDerived_PolyPhen2_0.5"

### 先处理del文件
dfdel <- read_tsv("data/004_delCount/delRatio.txt.gz") 

dftaxaDB <- read_tsv("data/001_TaxaDB/WheatVMap2_GermplasmInfo.txt") %>% 
  select(Taxa=VcfID,GenomeType,GroupbyContinent)

factorA <- c("AT","WE","DE","FTT","LR_EU","CL","LR_EA")
factorB <- c("Ratio_002_nonsynonymous","Ratio_031_GERP16way_1.5andDerived_PolyPhen2_0.5","Ratio_007_VEP")
labelsB <- c("Nonsynonymous","Deleterious","High impact")

df <- dfdel %>% left_join(.,dftaxaDB,by="Taxa") %>% 
  filter(DelCountGroup == "TotalDerivedSNPCount") %>% 
  select(-c("IfSelected","Group_refobj","DelCountGroup")) %>% 
  filter(GroupbyContinent %in% factorA) %>% 
  filter(Stand_LoadGroup %in% factorB) %>% 
  droplevels() %>% 
  mutate(GroupbyContinent= factor(GroupbyContinent, levels = factorA)) %>% 
  mutate(Stand_LoadGroup=factor(Stand_LoadGroup,levels = factorB, labels = labelsB))

colB <- c("#87cef9","#ffd702","#7f5701","#016699", "#fc6e6e","#9900ff","gray80")

```

### plot
```{r}
p <- df %>%
  ggplot(aes(x=Subgenome,y=Stand_delCount,fill=GroupbyContinent))+
  ylab("Mutation burden")+xlab("")+
  ### volin
  geom_violin(position = position_dodge(0.8),alpha=0.8,width=2) +
  geom_boxplot(position = position_dodge(0.8), width=0.01, color="white", alpha=0.2,outlier.color = NA) +
  # geom_point(position = position_jitterdodge(jitter.width = 0.5, dodge.width = 0.8),aes(color=GroupbyContinent),alpha=0.3)+
  stat_summary(fun=mean, geom="point", color="red",position = position_dodge(0.8),size=0.5)+

  ### boxplot
  # geom_boxplot(position = position_dodge(0.8),outlier.color = NA, alpha = 0.8) +
  # geom_point(position = position_jitterdodge(jitter.width = 0.5, dodge.width = 0.8),aes(color=GroupbyContinent),alpha=0.3)+
  # stat_summary(fun=mean, geom="point", color="red",position = position_dodge(0.8),size=0.5)+
  
  ### pointrange
  # stat_summary(aes(color=GroupbyContinent),fun.data=mean_sdl,position = position_dodge(0.8),geom="pointrange",size=0.3)+
  
  scale_color_manual(values = colB,name='')+
  scale_fill_manual(values = colB,name='')+
  facet_grid(Stand_LoadGroup~.,scales = "free")+
  # facet_wrap(Stand_LoadGroup~.,scales = "free", ncol = 1)+
  # ggtitle("Del total count under all genes in vmap 2.1-2021") +
  theme(strip.background = element_blank())+
  theme(
    panel.background = element_blank(),
    panel.border = element_rect(colour = "black", fill=NA, size=0.4)
    ,legend.position = 'none'
    ,text = element_text(size = 9))
p

# ggsave(p,filename = "~/Documents/test.pdf",height = 3,width = 4)
ggsave(p,filename = "figs/Fig3/misc/whole_load.pdf",height = 3,width = 4)
```

## Del count
```{r}
### 先处理del文件
dfdel <- read_tsv("data/004_delCount/delCount.txt.gz") 

dftaxaDB <- read_tsv("data/001_TaxaDB/WheatVMap2_GermplasmInfo.txt") %>% 
  select(Taxa=VcfID,GenomeType,GroupbyContinent)

factorA <- c("AT","WE","DE","FTT","LR_EU","CL","LR_EA")
factorB <- c("002_nonsynonymous","031_GERP16way_1.5andDerived_PolyPhen2_0.5","007_VEP")
labelsB <- c("Nonsynonymous","Deleterious","High impact")

df <- dfdel %>% left_join(.,dftaxaDB,by="Taxa") %>% 
  select(-c("IfSelected","Group_refobj","SiteCountWithMinDepth")) %>% 
  filter(GroupbyContinent %in% factorA) %>% 
  filter(VariantsGroup %in% factorB) %>% 
  droplevels() %>% 
  mutate(GroupbyContinent= factor(GroupbyContinent, levels = factorA)) %>% 
  mutate(VariantsGroup=factor(VariantsGroup,levels = factorB, labels = labelsB))

colB <- c("#87cef9","#ffd702","#7f5701","#016699", "#fc6e6e","#9900ff","gray70")
```


### plot
```{r}
p <- df %>%
  ggplot(aes(x=Subgenome,y=TotalDerivedSNPCount,fill=GroupbyContinent))+
  labs(y="No. derived SNPs",x="")+
  ### volin
  # geom_violin(position = position_dodge(0.8),alpha=0.8,width=2) +
  # geom_boxplot(position = position_dodge(0.8), width=0.01, color="white", alpha=0.2,outlier.color = NA) +
  # # geom_point(position = position_jitterdodge(jitter.width = 0.5, dodge.width = 0.8),aes(color=GroupbyContinent),alpha=0.3)+
  # stat_summary(fun=mean, geom="point", color="red",position = position_dodge(0.8),size=0.5)+

  ### boxplot
  # geom_boxplot(position = position_dodge(0.8),outlier.color = NA, alpha = 0.8) +
  # geom_point(position = position_jitterdodge(jitter.width = 0.5, dodge.width = 0.8),aes(color=GroupbyContinent),alpha=0.3)+
  # stat_summary(fun=mean, geom="point", color="red",position = position_dodge(0.8),size=0.5)+
  
  ### pointrange
  stat_summary(aes(color=GroupbyContinent),fun.data=mean_sdl,
               fun.args = list(mult = 1),
               position = position_dodge(0.8),geom="pointrange")+
  stat_summary(fun=mean, geom="point", color="white",position = position_dodge(0.8))+
  
  scale_color_manual(values = colB,name='')+
  scale_fill_manual(values = colB,name='')+
  facet_grid(VariantsGroup~.,scales = "free")+
  # facet_wrap(Stand_LoadGroup~.,scales = "free", ncol = 1)+
  # ggtitle("Del total count under all genes in vmap 2.1-2021") +
  theme(strip.background = element_blank())+
  theme(
    panel.background = element_blank(),
    panel.border = element_rect(colour = "black", fill=NA, size=0.4)
    ,legend.position = 'none'
    ,text = element_text(size = 9))
p

ggsave(p,filename = "~/Documents/test.pdf",height = 3,width = 4)
# ggsave(p,filename = "figs/Fig3/misc/whole_load_count.pdf",height = 3,width = 4)
```






### significance test - slightly
- 在分析显著性差异前，先将数据框赋予一个新的变量，存储下来。
```{r}
dfdf <- df
```
- 先计算平均值 标准差之类的
```{r}
### 平均数，标准差，方差等统计
  
df <- dfdf %>% filter(Stand_LoadGroup == "Nonsynonymous") %>% 
  mutate(GroupbyContinent = factor(GroupbyContinent)) %>% 
  group_by(Subgenome,GroupbyContinent) %>% 
summarise(n=n(),mean=mean(Stand_delCount),median=median(Stand_delCount),max=max(Stand_delCount),min=min(Stand_delCount),sd=sd(Stand_delCount), se = sd/sqrt(n))


df
write.table(df,file="/Users/Aoyue/Documents/value.txt",quote=F,col.name=T,row.names=F,sep = "\t") 

```


```{r}
### 写一个小函数，专门计算升高降低百分比
aochangePercent <- function(before, after){
  out <-  (after-before)/before
  print(paste("before",before,"\n","after",after,"out",out))
    return(out)
}

## we - de
print("we -de")
aochangePercent(df[1,4],df[2,4])
aochangePercent(df[7,4],df[8,4])
print("******************************")
## de-ftt
print("de -ftt")
aochangePercent(df[2,4],df[3,4])
aochangePercent(df[8,4],df[9,4])
print("******************************")
## ftt-lr 
print("ftt - lr")
aochangePercent(df[3,4],df[4,4])
aochangePercent(df[9,4],df[10,4])
print("******************************")
## ae-lr cl
print("ae - lr cl")
aochangePercent(df[13,4],df[14,4])
aochangePercent(df[13,4],df[15,4])
print("******************************")
## lr - cl
print("lr - cl")
aochangePercent(df[4,4],df[5,4])
aochangePercent(df[10,4],df[11,4])
aochangePercent(df[14,4],df[15,4])
print("******************************")

```


#### ====
## del by AB sub D sub
```{r}

```


#### =============
## XPCLR load ratio
```{r}
dfdelRatio <- read.delim("/Users/Aoyue/project/wheatVMap2_1000/002_dataAnalysis/009_xpclr/004_summary_XPCLR_top005/012_merge/delRatio.txt")

### 导入种质数据库,并选取特定的列
dftaxaDB <- read_tsv("data/001_TaxaDB/WheatVMap2_GermplasmInfo.txt") %>%
  select(Taxa=VcfID,GenomeType,GroupbyContinent,IfOnVmap2_S643)

df <- dfdelRatio %>% left_join(.,dftaxaDB,by="Taxa")
df$GroupbyContinent <- factor(df$GroupbyContinent, levels = c("WE","DE","FTT","LR_EU","CL"))
colB <- c("#ffd702","#7f5701","#016699", "#fc6e6e","#9900ff")

df$Stand_LoadGroup <- factor(df$Stand_LoadGroup,levels = c("Ratio_002_nonsynonymous","Ratio_031_GERP16way_1.5andDerived_PolyPhen2_0.5","Ratio_007_VEP"), labels = c("Nonsynonymous","Deleterious","High impact"))

df$Group_refobj <- factor(df$Group_refobj, levels = c("wede","dedurum","lrcul"))

p <- df %>% 
  filter(IfSelected == 1) %>% 
  filter(DelCountGroup == "TotalDerivedSNPCount") %>% 
  ggplot(aes(x=Subgenome,y=Stand_delCount))+
  ylab("Mutation burden")+xlab("")+
  ### pointrange
  stat_summary(aes(color=GroupbyContinent),fun.data=mean_sdl,
               fun.args = list(mult = 1),
               position=position_dodge(0.8),geom="pointrange")+
  stat_summary(aes(group = GroupbyContinent), fun=mean, geom="point", color="white",position = position_dodge(0.8))+
  scale_color_manual(values = colB,name='')+
  scale_fill_manual(values = colB,name='')+
  facet_grid(Stand_LoadGroup~Group_refobj,scales = "free")+
  theme(strip.background = element_blank())+
  theme(
    panel.background = element_blank(),
    panel.border = element_rect(colour = "black", fill=NA, size=0.7)
    ,legend.position = 'none'
    ,text = element_text(size = 12))
p

# ggsave(p,filename = "~/Documents/test.pdf",height = 3,width = 4)
ggsave(p,filename = "~/Documents/test.pdf",height =6,width = 6)

```

## XPCLR load count
```{r}
dfdelcount <- read_tsv("/Users/Aoyue/project/wheatVMap2_1000/002_dataAnalysis/009_xpclr/004_summary_XPCLR_top005/012_merge/delCount_xpclr.txt")

### 导入种质数据库,并选取特定的列
dftaxaDB <- read_tsv("data/001_TaxaDB/WheatVMap2_GermplasmInfo.txt") %>%
  select(Taxa=VcfID,GenomeType,GroupbyContinent)

factorA <- c("002_nonsynonymous","031_GERP16way_1.5andDerived_PolyPhen2_0.5","007_VEP")
labelsA <- c("Nonsynonymous","Deleterious","High impact")

df <- dfdelcount %>% left_join(.,dftaxaDB,by="Taxa") %>% 
  filter(VariantsGroup %in% factorA)

df$GroupbyContinent <- factor(df$GroupbyContinent, levels = c("WE","DE","FTT","LR_EU","CL"))
colB <- c("#ffd702","#7f5701","#016699", "#fc6e6e","#9900ff")
df$VariantsGroup <- factor(df$VariantsGroup,levels = factorA, labels = labelsA)
df$Group_refobj <- factor(df$Group_refobj, levels = c("wede","dedurum","lrcul"))

p <- df %>% 
  filter(IfSelected == 1) %>% 
  ggplot(aes(x=Subgenome,y=TotalDerivedSNPCount))+
  ylab("Derived allele count per accession")+xlab("")+
  ### pointrange
  stat_summary(aes(color=GroupbyContinent),fun.data=mean_sdl,
               fun.args = list(mult = 1),size=0.4,
               position=position_dodge(0.95),geom="pointrange")+
  stat_summary(aes(group = GroupbyContinent), fun=mean, geom="point", color="white",
               size=0.9,
               position = position_dodge(0.95))+
  scale_color_manual(values = colB,name='')+
  scale_fill_manual(values = colB,name='')+
  facet_grid(VariantsGroup~Group_refobj,scales = "free")+
  theme(strip.background = element_blank(),
        strip.text = element_text(size=6))+
  theme(
    panel.background = element_blank(),
    panel.border = element_rect(colour = "black", fill=NA, size=0.5)
    ,legend.position = 'none'
    ,text = element_text(size = 9))
p

# ggsave(p,filename = "~/Documents/test.pdf",height = 3,width = 4)
# ggsave(p,filename = "~/Documents/test.pdf",height = 2.4,width = 2.4)
ggsave(p,filename = "~/Documents/test.pdf",height = 3,width = 3)

```


#### ==============
## DelLoad only Hexaploid
```{r}
choiceA <- "Ratio_031_GERP16way_1.5andDerived_PolyPhen2_0.5"

### 先处理del文件
dfdel <- read_tsv("data/004_delCount/delRatio.txt.gz") 

dftaxaDB <- read_tsv("data/001_TaxaDB/WheatVMap2_GermplasmInfo.txt") %>% 
  select(Taxa=VcfID,GenomeType,GroupbyContinent)

factorA <- c("LR_EU","CL","LR_EA")

# factorB <- c("Ratio_002_nonsynonymous","Ratio_031_GERP16way_1.5andDerived_PolyPhen2_0.5","Ratio_007_VEP")
# labelsB <- c("Nonsynonymous","Deleterious","High impact")

factorB <- c("Ratio_031_GERP16way_1.5andDerived_PolyPhen2_0.5")
labelsB <- c("Deleterious")

df <- dfdel %>% left_join(.,dftaxaDB,by="Taxa") %>% 
  filter(DelCountGroup == "TotalDerivedSNPCount") %>% 
  select(-c("IfSelected","Group_refobj","DelCountGroup")) %>% 
  filter(GroupbyContinent %in% factorA) %>% 
  filter(Stand_LoadGroup %in% factorB) %>% 
  droplevels() %>% 
  mutate(GroupbyContinent= factor(GroupbyContinent, levels = factorA)) %>% 
  mutate(Stand_LoadGroup=factor(Stand_LoadGroup,levels = factorB, labels = labelsB))

colB <- c( "#fc6e6e","#9900ff","gray90")

```

### plot
```{r}
p <- df %>%
  ggplot(aes(x=Subgenome,y=Stand_delCount,fill=GroupbyContinent))+
  ylab("Mutation burden")+xlab("")+
  ### volin
  # geom_violin(position = position_dodge(0.8),alpha=0.8,width=2) +
  # geom_boxplot(position = position_dodge(0.8), width=0.02, color="white", alpha=0.2) +
  # geom_point(position = position_jitterdodge(jitter.width = 0.5, dodge.width = 0.8),aes(color=GroupbyContinent),alpha=0.3)+
  # stat_summary(fun=mean, geom="point", color="red",position = position_dodge(0.8),size=0.5)+

  ### boxplot
  geom_boxplot(position = position_dodge(0.8),outlier.color = NA, alpha = 0.8) +
  geom_point(position = position_jitterdodge(jitter.width = 0.5, dodge.width = 0.8),aes(color=GroupbyContinent),alpha=0.3)+
  stat_summary(fun=mean, geom="point", color="red",position = position_dodge(0.8),size=0.5)+
  
  ### pointrange
  # stat_summary(aes(color=GroupbyContinent),fun.data=mean_sdl,position = position_dodge(0.8),geom="pointrange",size=0.3)+
  
  scale_color_manual(values = colB,name='')+
  scale_fill_manual(values = colB,name='')+
  facet_wrap(Stand_LoadGroup~.,scales = "free", ncol = 1)+
  # ggtitle("Del total count under all genes in vmap 2.1-2021") +
  theme(strip.background = element_blank())+
  theme(
    panel.background = element_blank(),
    panel.border = element_rect(colour = "black", fill=NA, size=0.7)
    ,legend.position = 'none'
    ,text = element_text(size = 12))
p

# ggsave(p,filename = "~/Documents/test.pdf",height = 3,width = 4)
ggsave(p,filename = "figs/Fig3/misc/whole_load.pdf",height = 3,width = 4)
```


#### ===============
## Introgression
#### ==============
## Data important
```{r}
# table <- read_tsv("data/fromDaxing/fd/006_introgression_burden/001_popFdIntervalBurden/introgressionDonorBurden.txt.gz") %>% 
#   mutate(Sub=substring(Chr, 2,2)) %>% 
#   mutate(P2=factor(P2,levels = c("LR_AM","LR_AF","LR_CSA","LR_WA","LR_EU","CL","LR_EA"))) %>% 
#   mutate(P3=factor(P3,levels = c("WE","DE","FTT","AT"),labels = c("WE","DE","FTT","AT"))) 
# 
# df <- table %>% 
#   group_by(Chr,Start,End,P3) %>%
#   summarise(across(contains("introgressed"),.fns = sum, na.rm = T))
# 
# df1 <- df %>% 
#   mutate(IntrogressedSynTotalCount=ifelse(NumAccessionIntrogressed==0,NA,IntrogressedSynTotalCount),
#          IntrogressedNonTotalCount=ifelse(NumAccessionIntrogressed==0,NA,IntrogressedNonTotalCount),
#          IntrogressedDelTotalCount=ifelse(NumAccessionIntrogressed==0,NA,IntrogressedDelTotalCount)) %>% 
#   mutate(NonIntrogressedSynTotalCount=ifelse(NumAccessionNonIntrogressed==0,NA,NonIntrogressedSynTotalCount),
#          NonIntrogressedNonTotalCount=ifelse(NumAccessionNonIntrogressed==0,NA,NonIntrogressedNonTotalCount),
#          NonIntrogressedDelTotalCount=ifelse(NumAccessionNonIntrogressed==0,NA,NonIntrogressedDelTotalCount))
# 
# write_tsv(df1,"/Users/Aoyue/project/wheatVMap2_1000/002_dataAnalysis/RProject/R4VMap2FigsS1062/data/fromDaxing/fd/006_introgression_burden/001_popFdIntervalBurden/002_introgressionDonorBurden.txt.gz")



table <- read_tsv("data/fromDaxing/fd/006_introgression_burden/001_popFdIntervalBurden/002_introgressionDonorBurden.txt.gz") %>% 
  mutate(Sub=str_sub(Chr,2,2)) %>% 
  mutate(Sub=factor(Sub, levels = c("A","B","D")), SubID=str_sub(Chr,1,1)) %>% 
  mutate(IntrogressedWeaklyLoad=(IntrogressedNonTotalCount+IntrogressedDelTotalCount)/IntrogressedSynTotalCount) %>%
  mutate(IntrogressedStronglyLoad=(IntrogressedDelTotalCount)/IntrogressedSynTotalCount) %>% 
  mutate(NonIntrogressedWeaklyLoad=(NonIntrogressedNonTotalCount+NonIntrogressedDelTotalCount)/ NonIntrogressedSynTotalCount) %>% 
  mutate(NonIntrogressedStronglyLoad=(NonIntrogressedDelTotalCount)/NonIntrogressedSynTotalCount)

### 结果里有 NA  NaN Inf
dfMaxMin <- table %>% 
  pivot_longer(cols = IntrogressedWeaklyLoad:NonIntrogressedStronglyLoad,names_to = "Group",values_to = "Load") %>% 
  filter(!is.infinite(Load), !is.na(Load)) %>% 
  group_by(Sub,P3,Group) %>% 
  summarise(max=max(Load))

df <- table %>% 
  pivot_longer(cols = IntrogressedWeaklyLoad:NonIntrogressedStronglyLoad,names_to = "Group",values_to = "Load") %>% 
  left_join(y = dfMaxMin,by = c("Sub","P3","Group")) %>% 
  mutate(Load=ifelse(Load>max, max, Load)) %>%  #### 对缺失值的处理
  select(-c("max")) %>% 
  pivot_wider(names_from = "Group", values_from = "Load") %>% 
  ## Normalization
  mutate(WeaklyLoadNew2=(IntrogressedWeaklyLoad-NonIntrogressedWeaklyLoad)/(IntrogressedWeaklyLoad+NonIntrogressedWeaklyLoad),
         StronglyLoadNew2=(IntrogressedStronglyLoad-NonIntrogressedStronglyLoad)/(IntrogressedStronglyLoad+NonIntrogressedStronglyLoad)) %>% 
  pivot_longer(cols = c(WeaklyLoadNew2,StronglyLoadNew2),names_to = "SlightlyOrStrongly",values_to = "IntrogressionBurdenIndex") %>% 
  mutate(SlightlyOrStrongly=factor(SlightlyOrStrongly, levels = c("WeaklyLoadNew2","StronglyLoadNew2"), labels = c("Nonsynonymous","Deleterious")))

df$P3 <- factor(df$P3,levels = c("AT","WE","DE","FTT"))
colB <- c("#87cef9","#ffd702","#7f5701","#016699")
```


##  load ratio (Introgressed/Nonintrogressed) IntrogressionBurdenIndex
```{r}

p <- df %>% 
  ggplot(aes(x=Sub, y=IntrogressionBurdenIndex, fill=P3))+
  # geom_boxplot()+
  # geom_lv(position = position_dodge(0.8),width.method = "height",color="black",conf=0.9,k=4)+
  # stat_summary(geom = "point",fun = mean,size=1, color="black",
  #              position = position_dodge(0.8))+
  geom_hline(yintercept = 0, linetype=2, color="grey")+
  stat_summary(geom = "pointrange", fun.data = mean_cl_boot, aes(color=P3), 
               position = position_dodge(0.8))+
  labs(y="Introgressed VS\nnonintrogressed burden",x="")+
  # scale_fill_brewer(palette = "Set2")+
  # scale_color_brewer(palette = "Set2")+
  scale_fill_manual(values = colB)+
  scale_color_manual(values = colB)+
  # scale_y_log10(breaks = scales::trans_breaks("log2", function(x) 2^x),
  #               labels = scales::trans_format("log2", scales::math_format(2^.x)))+
  # coord_cartesian(xlim = c(0,1.0))+
  facet_grid(SlightlyOrStrongly~.)+
  theme(
    axis.title.x = element_blank(),
    strip.background = element_blank(),
    panel.background = element_blank(),
    panel.border = element_rect(colour = "black", fill=NA, size=0.5),
    legend.position = 'none',
    text = element_text(size = 9))
p

ggsave(p,filename = "~/Documents/test.pdf",height = 2.4,width = 2.4)


dftest <- df %>% 
  filter(!is.na(IntrogressionBurdenIndex)) %>% 
  group_by(P3,SlightlyOrStrongly) %>% 
  count()

dftest2 <- df %>% 
  # filter(is.na(IntrogressionBurdenIndex)) %>% 
  group_by(P3,SlightlyOrStrongly) %>% 
  count()

```

## whole genome distribution
```{r}
dfall <- df
```

```{r}
library(slider)
df <- dfall %>% 
  filter(!is.na(IntrogressionBurdenIndex)) %>% 
  group_by(Chr,P3) %>% 
  mutate(bin=slide(Start,mean,.before = 50,.after = 50) %>% unlist(),
         binBurdenIndex=slide(IntrogressionBurdenIndex, mean,.before = 50,.after = 50) %>% unlist())

#### 设置每条染色体的点
data1 <- data.frame(RefChr = paste(1:7,rep("A",7),sep = ""),x=c(213,327,319,266,254,286,362),y=rep(0,7))
data2 <- data.frame(RefChr = paste(1:7,rep("B",7),sep = ""),x=c(241,348,347,304,201,325,309),y=rep(0,7))
data3 <- data.frame(RefChr = paste(1:7,rep("D",7),sep = ""),x=c(170,268,240,185,187,215,339),y=rep(0,7))
data <- rbind(data1,data2,data3) %>% 
  mutate(Sub=str_sub(RefChr,2,2), SubID=str_sub(RefChr,1,1)) %>% 
  filter(RefChr %in% c("1A","1B","1D"))

p <- df %>%
  group_by(SlightlyOrStrongly) %>% 
  # filter(Chr %in% c("1A","1B","1D")) %>% 
  group_map(~{ggplot(.x,aes(x=bin/1000000,y=binBurdenIndex,color=P3))+
  # geom_point()+
  geom_line(size=0.4)+
  geom_hline(yintercept = 0, linetype=2, color="grey",size=0.4)+
  facet_grid(SubID~Sub)+
  geom_point(aes(x,y),color = "blue",size=0.5,data = data)+
  scale_fill_manual(values = colB)+
  scale_color_manual(values = colB)+
  scale_y_continuous(breaks = seq(-0.3,0.1,by=0.3))+
  # geom_smooth(method = "loess",span=0.2)+
  labs(x="Genomic position (Mb)", y="Introgressed VS 
       nonIntrogressed burden")+
  theme_classic()+
  theme(
        legend.background = element_rect(fill="transparent"),
        strip.background = element_blank(),
        legend.position = "none",
        legend.title = element_blank(),
        # axis.title.x = element_blank(),
        axis.line = element_line(size = 0.3),
        text = element_text(size = 9),
        panel.border = element_rect(colour = "transparent",fill = "transparent"))})
p
ggsave("~/Documents/temp.pdf",p[[1]],width =4.73 , height = 2.31)


p <- df %>%
  filter(P3=="AT") %>%
  ggplot(aes(x=bin/1000000,y=binBurdenIndex,color=SlightlyOrStrongly))+
  # geom_point()+
  geom_line()+
  geom_point(aes(x,y),color = "blue",size=1,data = data)+
  geom_hline(yintercept = 0, linetype=2, color="grey")+
  facet_grid(SubID~Sub)+
  scale_fill_manual(values = colB)+
  scale_color_manual(values = colB)+
  labs(x="Genomic position (Mb)")+
  # geom_smooth(method = "loess",span=0.2)+
  theme_classic()+
  theme(
        legend.background = element_rect(fill="transparent"),
        strip.background = element_blank(),
        # legend.position = "none", legend.title = element_blank(),
        axis.title.x = element_blank(),
        axis.line = element_blank(),
        # text = element_text(size = 14),
        panel.border = element_rect(colour = "black",fill = "transparent"))

p
ggsave(p,filename = "~/Documents/test.pdf",height = 7.2,width = 7.2)
  
```



##  

```{r}

```






























 



